processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByQueryEngineV2.java

```diff
@@ -342,6 +350,92 @@ private void initializeGroupingKeyV2Dimension(
     }
   }
 
+  private static class LongGroupByColumnSelectorStrategy implements GroupByColumnSelectorStrategy
+  {
+    @Override
+    public int getGroupingKeySize()
+    {
+      return Longs.BYTES;
+    }
+
+    @Override
+    public void processValueFromGroupingKey(
+        GroupByColumnSelectorPlus selectorPlus, ByteBuffer key, Map<String, Object> resultMap
+    )
+    {
+      final long val = key.getLong(selectorPlus.getKeyBufferPosition());
+      resultMap.put(selectorPlus.getOutputName(), val);
+    }
+
+    @Override
+    public void initColumnValues(ColumnValueSelector selector, int columnIndex, Object[] valuess)
+    {
+      valuess[columnIndex] = ((LongColumnSelector) selector).get();
+    }
+
+    @Override
+    public void initGroupingKeyColumnValue(
+        int keyBufferPosition, int columnIndex, Object rowObj, ByteBuffer keyBuffer, int[] stack
+    )
+    {
+      keyBuffer.putLong(keyBufferPosition, (Long) rowObj);
+      stack[columnIndex] = 1;
+    }
+
+    @Override
+    public boolean checkRowIndexAndAddValueToGroupingKey(
+        int keyBufferPosition, Object rowObj, int rowValIdx, ByteBuffer keyBuffer
+    )
+    {
+      // rows from a long column are always size 1
```

Comment:
I've reworded the comment to mention multivalue support, and replaced the constant with a named one